lib/spack/llnl/util/tty/color.py

```diff
@@ -263,7 +263,11 @@ def match_to_ansi(match):
                 f"Incomplete color format: '{match.group(0)}' in '{match.string}'"
             )
 
-        ansi_code = _escape(f"{styles[style]};{colors.get(color_code, '')}", color, enclose, zsh)
+        color_number_str = ""
+        color_number = colors.get(color_code)
+        if color_number:
+            color_number_str = f";{color_number}"
+        ansi_code = _escape(f"{styles[style]}{color_number_str}", color, enclose, zsh)
```

Comment:
Trying to avoid an extra format:
```suggestion
        color_number = colors.get(color_code, "")
        semi = ";" if color_number else ""
        ansi_code = _escape(f"{styles[style]}{semi}{color_number}", color, enclose, zsh)
```